### DISCUSSION TITLE
Update Traefik

### DISCUSSION THREAD

**Discussion by pelifix** [rancher-sandbox/rancher-desktop#6468] [Q&A] [answered]
Hi all!

How can I update the version of _Traefik_ that is running in _Rancher Desktop_?
I'm running Rancher Desktop 1.12.2 (on Win 11) and I have Traefik version 2.9.10.

thx!

--
christer

**Comment by jandubois** [comment, marked as answer]
The version of Traefik being used depends on the Kubernetes version you are running; it is bundled with the k3s-airgap images.

If you want to run a different version, then you need to [disable Traefik]( https://docs.rancherdesktop.io/ui/preferences/kubernetes) in Rancher Desktop and deploy the version you want using `helm`.

**Comment by pelifix** [comment]
Thanks @jandubois , that makes sense.

How can I find out which Traefik version is bundled with Kubernetes 1.28.x  (or k3s-airgap)?
Update:
I think I found it. The k3s-airgap release [v1.28.3+k3s1](https://github.com/k3s-io/k3s/releases/tag/v1.28.3%2Bk3s1)  is bundled with Traefik v2.10.5, and is the release I need. Will try to upgrade k8s to 1.28.3 or newer, that should fix my issue.

thanks!

**Reply by jandubois** [reply]
Exactly, you can find it in the `k3s` release notes for each version.

**Reply by jandubois** [reply]
Just want to add: Rancher Desktop is **not** using the `containerd` and `runc` versions bundled by `k3s`, but supplies it's own. That way we run even old `k3s` versions on the newer `containerd` and `runc` releases that include the latest security updates.

But everything that runs inside the Kubernetes cluster comes from the `k3s` tarballs.
